### PR TITLE
Gradient optimizer.

### PR DESCRIPTION
**Pull request by joaoamcarvalho**

### CODE REVIEW

**Comment by joaoamcarvalho** [comment]
Uses a gradient optimizer class for policy gradients and black box optimization.

**Review by davide-tateo90 — COMMENTED** [review]
- change learning_rate to optimizer
- omega -> omega_old
- grad_optimizer to optimizer

**Comment by joaoamcarvalho** [comment]
@boris-il-forte Check the changes and merge.

**Review by davide-tateo90 — COMMENTED** [review]
- Fix helper functions for other optimizers
- Change AdaptiveParameterOptimizer to AdaptiveOptimizer

**Review by davide-tateo90 — COMMENTED** [review]
- rename FixedLearningRateOptimizer into SGDOptimizer
- value -> learning rate

**Review by davide-tateo90 — COMMENTED** [review]
- simplify call/update/get value of AdaptiveOptimizer
- set learning rate to parameter if float
